### ISSUE TITLE
[Clang] bogus error about vector values of different size

### ISSUE THREAD

**Issue by philnik777** [llvm/llvm-project#161360]
With `-flax-vector-conversions=none`, clang rejects the following code:
```c++
template <class T>
using vec [[clang::ext_vector_type(4)]] = T;

auto test(vec<int> v, vec<const int> v2) {
  return v == v2;
}
```

I'm not quite sure whether it's intended to work, but the diagnostic is definitely bogus:
```
<source>:6:12: error: cannot convert between vector values of different size ('vec<int>' (vector of 4 'int' values) and 'vec<const int>' (vector of 4 'const int' values))
    6 |   return v == v2;
      |          ~ ^  ~~
```
Both the element size and total vector size are identical.

**Comment by tbaederr** [comment]
This is emitted from the end of `Sema::CheckVectorOperands`.

The `LHSType`/`RHSType` are:
```console
TemplateSpecializationType 0x7dab2fc57740 'vec<int>' sugar alias
|-name: 'vec' qualified
| `-TypeAliasTemplateDecl 0x7dab2fc57598  vec
|-TemplateArgument type 'int'
| `-BuiltinType 0x7dab2fbfe270 'int'
`-ExtVectorType 0x7dab2fc576f0 'int __attribute__((ext_vector_type(4)))' 4
  `-SubstTemplateTypeParmType 0x7dab2fc57650 'int' sugar class depth 0 index 0 T final
    |-TypeAliasTemplate 0x7dab2fc57598 'vec'
    `-BuiltinType 0x7dab2fbfe270 'int'

TemplateSpecializationType 0x7dab2fc579a0 'vec<const int>' sugar alias
|-name: 'vec' qualified
| `-TypeAliasTemplateDecl 0x7dab2fc57598  vec
|-TemplateArgument type 'const int'
| `-QualType 0x7dab2fbfe271 'const int' const
|   `-BuiltinType 0x7dab2fbfe270 'int'
`-ExtVectorType 0x7dab2fc57950 'const int __attribute__((ext_vector_type(4)))' 4
  `-SubstTemplateTypeParmType 0x7dab2fc578b0 'const int' sugar class depth 0 index 0 T final
    |-TypeAliasTemplate 0x7dab2fc57598 'vec'
    `-QualType 0x7dab2fbfe271 'const int' const
      `-BuiltinType 0x7dab2fbfe270 'int'
```
